Shivani_Bansal/calculator/Calculator.js

```diff
@@ -0,0 +1,102 @@
+function insert(num){
+    // calculator.display.value += num
+    console.log(num,num.target, "ab")
+
+    document.getElementById("display").value = document.getElementById("display").value + num;
+    
+}
+
+function solve(){
+    txt = document.getElementById("display").value;
+    l = txt.length;
+    
+    numb = txt.slice(4, l-1); 
+
+    // var pos = txt.indexOf("√") ;
+    // txt = txt[0, pos] + "Math.sqrt(" + 
+
+    // if (txt.startsWith("√")) {
+
+    //     document.getElementById("display").value = Math.sqrt(txt.slice(1));
+    //     }
+
+    if (txt.charAt(l-1)=="!") {
+        ans = Factorial(txt.slice(0, l-1));
+        document.getElementById("display").value = ans
+        
+    }
+    else{
+    txt = txt.replace(/√/g, "Math.sqrt");
+    txt = txt.replace(/Sin/g, "Math.sin");
+    txt = txt.replace(/Sin/g, "Math.sin");
+    txt = txt.replace(/Cos/g, "Math.cos");
+    txt = txt.replace(/Tan/g, "Math.tan");
+    txt = txt.replace(/X/g, "*");
+    txt = txt.replace(/÷/g, "/");
+    ans= eval(txt);
```

Comment:
try to find a workaround to make this work without using eval(), eval is degrading performance and it is not recommended to use as it can break the natural execution context creation and introduce dynamically-scoped issues